dev/fattest.simplicity/src/componenttest/rules/repeater/MicroProfileActions.java

```diff
@@ -536,4 +543,40 @@ public static RepeatTests repeat(String[] servers, TestMode otherFeatureSetsTest
     public static RepeatTests repeat(String[] servers, TestMode otherFeatureSetsTestMode, boolean skipTransformation, FeatureSet firstFeatureSet, FeatureSet... otherFeatureSets) {
         return RepeatActions.repeat(servers, otherFeatureSetsTestMode, ALL, firstFeatureSet, Arrays.asList(otherFeatureSets), skipTransformation);
     }
+
+    /**
+     * Get a repeat test for the following FeatureSets, with a predicate that will be used to filter out the FeatureSets.
+     * The first FeatureSet to pass the predicate will be run in LITE mode. The others will be run in FULL.
+     *
+     * @param  serverName  the server to repeat on
+     * @param  predicate   a predicate that will filter feature sets. If the predicate returns true, the feature set will be run
+     * @param  featureSet  the first feature set
+     * @param  featureSets the feature sets
+     * @return             a RepeatTests instance
+     */
+    public static RepeatTests repeatIf(String serverName, Predicate<FeatureSet> predicate, FeatureSet featureSet, FeatureSet... featureSets) {
+
+        final String m = "repeatIf";
+
+        List<FeatureSet> featureSetsMerged = new ArrayList<FeatureSet>(Arrays.asList(featureSets));
+        featureSetsMerged.add(0, featureSet);
+
+        Log.info(c, m, "enter. Testing " +
+                       String.join(", ", featureSetsMerged.stream().map(FeatureSet::getID).collect(Collectors.toList())));
```

Comment:
There's actually a collector that builds the string directly.

```suggestion
        Log.info(c, m, "enter. Testing " +
                       featureSetsMerged.stream().map(FeatureSet::getID).collect(Collectors.joining(", ")));
```